phoenix-shared/src/commonMain/kotlin/fr.acinq.phoenix/data/Wallet.kt

```diff
@@ -34,6 +34,28 @@ data class Wallet(val seed: ByteArray) {
         return Bitcoin.computeBIP84Address(publicKey, chainHash)
     }
 
+    // For cloud storage, we need:
+    // - an encryption key
+    // - a cleartext string that's tied to a specific nodeId
+    //
+    // But we also don't want to expose the nodeId.
+    // So we deterministically derive both values from the seed.
+    //
+    fun cloudKeyAndEncryptedNodeId(chain: Chain): Pair<ByteVector32, String> {
+        val isTestnet = when (chain) {
+            is Chain.Testnet, is Chain.Regtest -> true
+            else -> false
+        }
+        val path = if (isTestnet) "m/48'/0'/100'/0" else "m/50'/0'/100'/0"
```

Comment:
The derivation path should use the same first level (50 for example) and then use the second one to discriminate between the chain, probably with `0` for mainnet and `1` for testnet. Let's summon @sstone for this!

Reply:
There are no real rules and we could use any derivation path but the convention for wallets (see https://en.bitcoin.it/wiki/BIP_0044 for example) is 
`m / purpose' / coin_type' /...`, `coin_type` being 0 for mainnet and 1 for testnet.
If we use this convention we should avoid picking values for `purpose` that are already used (44', 49', 84').

To be honest though, it is not the convention we used for eclair's internal keys (we chose 46'/1' for mainnet and 47'/1' for testnet) :)

Reply:
I don't have any opinions here, and I'm fine with the proposal:
```
val path = if (isTestnet) "m/50'/1'/0'/0" else "m/50'/0'/0'/0"
```
It does look like we're already using `m/50'/0'/...` in `LocalKeyManager.nodeKeyBasePath`: [link](https://github.com/ACINQ/lightning-kmp/blob/aa1772878bdbd98ec4b93e60fa729de8d4dc3300/src/commonMain/kotlin/fr/acinq/lightning/crypto/LocalKeyManager.kt#L133). I was trying to avoid re-using any existing derivation path. What do we think ?

Reply:
How about `if (isTestnet) "m/51'/1'/0'/0" else "m/51'/0'/0'/0"`?

Reply:
Done in 2ea3d48 
```
val path = if (isTestnet) "m/51'/1'/0'/0" else "m/51'/0'/0'/0"
```